Source/GUI/MemViewer/MemViewer.cpp

```diff
@@ -131,25 +121,85 @@ void MemViewer::changeMemoryRegion(const bool isMEM2)
 {
   m_memViewStart = isMEM2 ? Common::MEM2_START : Common::MEM1_START;
   m_memViewEnd = isMEM2 ? Common::MEM2_END : Common::MEM1_END;
-  verticalScrollBar()->setRange(0, ((m_memViewEnd - m_memViewStart) / 16) - 1);
+  verticalScrollBar()->setRange(0, ((m_memViewEnd - m_memViewStart) / VISIBLE_COLS) - VISIBLE_ROWS);
 }
 
 void MemViewer::mousePressEvent(QMouseEvent* event)
 {
+  // Only handle left-click events
+  if (event->button() != Qt::MouseButton::LeftButton)
+    return;
+
   int x = event->pos().x();
   int y = event->pos().y();
-  if (x >= m_rowHeaderWidth && x <= m_rowHeaderWidth + m_hexAreaWidth && y >= m_columnHeaderHeight)
+
```

Comment:
Removing this actually doesn't work, you can select the first of the row if you click on the last hex character  of the row header and it really does not ignore clicks on the header (`clickedPosY` is calculated using integer division while the left member, if you click on the column header, will always be equal or less then the right member which means, it's always 0 here).  After testing a bit, this is the correct way to do this:

```
if (!(x >= m_rowHeaderWidth &&
        x <= m_rowHeaderWidth + m_hexAreaWidth + spacing + (m_charWidthEm * VISIBLE_COLS) &&
        y >= m_columnHeaderHeight && y <= m_columnHeaderHeight + m_charHeight * VISIBLE_ROWS))
    return;
```

It assumes `spacing` is declared a line above and obviously, replace the define constant by whatever name you give to the class wide `const`.  Note also the addition of `&& y <= m_columnHeaderHeight + m_charHeight * VISIBLE_ROWS`, this is because while testing this, I found a bug that is affecting master where if you click JUST RIGHT you can select an invisible row at the bottom, I found this PR fitting for the fix so I put it in there.  Essentially, this condition ensure that whatever you clicked in is in the hex area or the ascii area, the entire rectangle.

You are free to put a comment there if it's confusing, but if you do go with this, remove the test you do on `clickedPosY`.

Reply:
Well spotted. Clicking the row, column headers or area between the two tables was due to the devision. I didn't take into account that any value between 0 and -width would return 0. I think a nicer way than having those concatinated conditions is to check if the relative click position is < 0 rather than checking the index.